### PR TITLE
Do not write to the read-only paths if system-buildpacks are provided

### PR DESCRIPTION
**Pull request by pbusko**

### CODE REVIEW

File: cmd/builder/cli/cli.go [hunk 1]
```diff
@@ -118,7 +132,7 @@ var builderCmd = &cobra.Command{
 			return errors.ErrGenericBuild
 		}
 
-		buildpackList, err = buildpacks.Translate(buildpackList, buildpacksDir, logger)
+		buildpackList, err = buildpacks.Translate(buildpackList, systemBuildpacksDir, logger)
 		if err != nil {
 			logger.Errorf("failed to translate buildpack locations %#v, error: %s\n", buildpackList, err.Error())
```
Comment (c0d1ngm0nk3y) [inline cmd/builder/cli/cli.go:137]: The logging should also be adapted, right?

File: cmd/builder/cli/cli.go [hunk 2]
```diff
@@ -87,7 +87,21 @@ var builderCmd = &cobra.Command{
...
+		tempDirs := map[string]*string{
+			"platform":       &platformDir,
+			"buildpacks":     &buildpacksDir,
+			"extensions":     &extensionsDir,
+			"download-cache": &downloadCacheDir,
+		}
+
+		for name, dir := range tempDirs {
+			if *dir, err = os.MkdirTemp("", name); err != nil {
+				logger.Errorf("failed to create %s folder, error: %s\n", name, err.Error())
```
Comment (c0d1ngm0nk3y) [inline cmd/builder/cli/cli.go:99]:
```suggestion
				logger.Errorf("failed to create folder %q, error: %s\n", name, err.Error())
```

File: cmd/builder/cli/cli.go [hunk 3]
```diff
@@ -62,6 +61,7 @@ func Execute() error {
 
 func init() {
 	builderCmd.Flags().StringSliceVarP(&buildpackList, "buildpack", "b", nil, "buildpack(s) to use")
+	builderCmd.Flags().StringVarP(&systemBuildpacksDir, "system-buildpacks-dir", "", "/tmp/buildpacks", "system buildpacks dir")
```
Comment (c0d1ngm0nk3y) [inline cmd/builder/cli/cli.go:64]: Why not `os.TempDir()`?

Comment (pbusko) [reply cmd/builder/cli/cli.go:64]: `system-buildpacks-dir` should point to the directory created by Diego, when it mounts system-buildpacks. This folder is read-only.

File: cmd/builder/cli/cli.go (hunk 1 above)
Comment (pbusko) [reply cmd/builder/cli/cli.go:137]: the logic remains the same.
before this change we used the same folder for downloading buildpacks and searching for system-buildpacks. the only thing that's changes is that these two folders are now different

Comment (c0d1ngm0nk3y) [reply cmd/builder/cli/cli.go:137]: My bad. I misread - I thought the directory is part of the error message.

File: cmd/builder/cli/cli.go (hunk 3 above)
Comment (c0d1ngm0nk3y) [reply cmd/builder/cli/cli.go:64]: Got it.

**Review by c0d1ngm0nk3y — APPROVED** [review]

**Pull request merged**